### ISSUE TITLE
Batch size for training

### ISSUE THREAD

**Issue by tlapusan** [yhenon/keras-frcnn#67]
Hi @yhenon ,
I saw that the batch size got from data_gen_train.next() is 1.
Playing with other networks, I figure out that a bigger batch size (ex. 64) helps the network to converge better and faster.

Do you think that a bigger batch size could help in this case ?

Thanks,
Tudor

**Comment by jahdiel** [comment]
@yhenon I would also apreciate if this was answered.

**Comment by yuayi521** [comment]
@tlapusan  I also find this issue, and I use fit_generator() function

**Comment by shahruk10** [comment]
@yuayi521  did using fit_generator() make training faster?

**Comment by yuayi521** [comment]
no, use fit_generator() can implement batch training, the author used fit() function with batch is 1.

**Comment by giladdiv** [comment]
The loss for the classification is 
`def class_loss_cls(y_true, y_pred):
return lambda_cls_class * K.mean(categorical_crossentropy(y_true[0, :, :], y_pred[0, :, :]))`
so i don't think it supports batch training

**Comment by ftiguidou** [comment]
@yuayi521, can you share a bit about how you used the fit_generator() function ? I want to try this way but I am wondering how you managed to do it. Did you totally separate the RPN training from the classifier training ? How did you sustain the process between the output of your model_rpnn.fit_generator and the input of the model_classifier training ? Moreover, as @giladdiv said, is the loss function already adapted for batch training ?
